src/packages/@ncigdc/modern_components/BiospecimenCard/BioTree/index.js

```diff
@@ -0,0 +1,240 @@
+  setDisplayName('EnhancedBioTreeBranch'),
+  withProps(({
+    query = '',
+  }) => ({
+    queryLowerCased: query.toLowerCase(),
+  })),
+  withState('emitterToken', 'setEmitterToken', null),
+  withState('expanded', 'setExpanded', ({
+    parentNode,
+    treeStatusOverride,
+  }) => (
+    ['expanded', 'query matches'].includes(treeStatusOverride) ||
+    parentNode === 'root'
+  )),
+  withHandlers({
+    branchClickHandler: ({
+      expanded,
+      setExpanded,
+      setTreeStatus,
+      setTreeStatusOverride,
+    }) => event => {
+      event.stopPropagation();
+      setExpanded(!expanded);
+      setTreeStatus({
+        type: `${expanded
+          ? 'COLLAPSE'
+          : 'EXPAND'}_NODE`,
+      });
+      setTreeStatusOverride();
+    },
+  }),
+  withPropsOnChange(
+    (
+      {
+        queryLowerCased,
+        treeStatusOverride,
+      },
+      {
+        queryLowerCased: nextQueryLowerCased,
+        treeStatusOverride: nextTreeOverride,
+      }
+    ) => !(
+      queryLowerCased === nextQueryLowerCased &&
+      treeStatusOverride === nextTreeOverride
+    ),
+    ({
+      entities,
+      expanded,
+      queryLowerCased,
+      search,
+      setExpanded,
+      setTreeStatus,
+      setTreeStatusOverride,
+      treeStatusOverride,
+      type,
+    }) => {
+      if (queryLowerCased.length > 0 && (
+        entities.hits.edges.some(e => search(queryLowerCased, e).length) ||
+        [
+          'samples',
+          'portions',
+          'analytes',
+          'aliquots',
+          'slides',
+        ].find(t => t.includes(queryLowerCased)) ||
+        type.p.includes(queryLowerCased)
+      )) {
+        const message = 'query matches';
+        setExpanded(true);
+        setTreeStatus({
+          payload: {
+            expanded: message,
+          },
+          type: 'OVERRIDE_NODES',
+        });
+        setTreeStatusOverride(message);
+      } else if (treeStatusOverride) {
+        const override = treeStatusOverride === 'expanded';
+        expanded === override || setExpanded(override);
+      }
+    }
+  ),
+  lifecycle({
+    componentDidMount() {
+      const {
+        expanded,
+        setTreeStatus,
+      } = this.props;
+
+      setTreeStatus({
+        payload: {
+          expanded,
+          mounted: true,
+        },
+        type: 'TOTAL_NODES',
+      });
+    },
+    componentWillMount(): void {
+      const {
+        setEmitterToken,
+        setExpanded,
+      } = this.props;
+      const token = Emitter.addListener('expand', toExpand => {
+        setExpanded(toExpand);
+      });
+
+      setEmitterToken(token);
+    },
+    componentWillUnmount(): void {
+      const {
+        emitterToken,
+        expanded,
+        setTreeStatus,
+      } = this.props;
+
+      emitterToken.remove();
+      setTreeStatus({
+        payload: {
+          expanded,
+          mounted: false,
+        },
+        type: 'TOTAL_NODES',
+      });
+    },
+    // shouldComponentUpdate({
```

Comment:
can this be deleted

Reply:
I've only included it as I had planned on adding enhancements on it later, but you're right... I'll add it when/if the time comes 👍